### ISSUE TITLE
Temporary files and log removal from c++ side

### ISSUE THREAD

**Issue by allegroCoder** [tuura/scenco#36]
Trying to remove as many temporary files as possible in scenco-cpp, as well as the log file.
In the next comment a list of temporary files and of their usage.

**Comment by snowleopard** [comment]
Thanks for the detailed summary, and for removing several files already!

Let me ask a couple of questions about the remaining ones:

> `TMP_FILE` is used for various reason.

Let's avoid vague statements like this :-) Please list all its use-cases, and probably it would be useful if they have separate check-boxes, so we can track them separately. Just edit the list above.

> `SCRIPT_TMP` dinamically store the script which is currently input into ABC. 

Can't you pass the script to ABC via standard input?

> `BOOL_PATH` store the Boolean equations generated by ABC for each node/arc.

Can't you capture ABC Boolean equations via standard output?

**Comment by allegroCoder** [comment]
> Just edit the list above.

Done.

> Can't you pass the script to ABC via standard input?

I have been trying now in Linux but I did not manage yet. `ABC` runs interactevely, therefore the only way I could make this execute a bunch of commands is through a script. Not sure if feeding `ABC` through stdin might be better than having a script though, considering compatibility issues between UNIX / OS X / Windows.

> Can't you capture ABC Boolean equations via standard output?

No, I have also been trying this now. But the only command I found for the Boolean equations generation is the following:
```text
write_eqn – Outputs the combinational part of the current network in the Synopsys equation format.
```
and it must be followed by a `file name`.

**Comment by snowleopard** [comment]
@allegroCoder Thanks! So, it looks like we can't use standard I/O for anything else.

Perhaps, the only way to get rid of temporary files would be to integrate ABC source code into `scenco`, similar to how this is done in `abcBridge`. But this will require quite a lot of work.

So, if you think we can't do anything else about the remaining temporary files now, feel free to close this.

**Comment by snowleopard** [comment]
Or we could keep this issue open, until we figure out a better way :-)

**Comment by allegroCoder** [comment]
Agree, let's keep this open and let's give it a think. Maybe we will come back to this later :+1:
